docs/developer/release-team.md

```diff
@@ -0,0 +1,80 @@
+
+This document introduces the concept of a release team with the following goals and non-goals:
+
+### Goals
+
+- To improve communication to end users and contributors about CAPI release cadence and target release dates
+- To spread the load of release tasks and involve a bigger, more diverse set of CAPI contributors in cutting releases
+- To look at the Kubernetes SIG release team processes for guidance on releasing best practices
+- To improve tooling, documentation, and automation of the CAPI release process
+
+### Non-Goals/Future work
+
+- To increase the frequency of releases (AKA release cadence). This will be revisited in the future once the release process stabilizes.
+- To change the current proposal (CAEP) process
+- To copy implement all steps of the Kubernetes release process for CAPI releases
+
+Note that this document is intended to be a starting point for the release team. It is not a complete release process document.
+
+More details on the CAPI release process can be found in [this past issue tracking release tasks](https://github.com/kubernetes-sigs/cluster-api/issues/6615) and the current [release documentation](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/developer/releasing.md).
+
+## Duration of Term
+
+Each release team term will last approximately four months, to align with one minor release cycle. A minor release cycle starts right after a minor release and concludes with the release of the next minor release. There is no limit to the number of terms a release team member can serve, meaning that it's possible for a release team member to serve multiple consecutive terms.
+
+As noted above, making changes to  the CAPI release cadence is out of scope for this initial release team process. 
+
+## Specific Responsibilities
+
+- Release patch releases monthly or more often as needed, so users will get fixes & updated dependencies with CVE fix on a predictable cadence
+- Release a minor release every 4 months (to be revised, see future work), so users will get new features on a predictable cadence
+- Create beta and rc tags for upcoming minor releases
+- Ensure eligible PRs are cherry-picked to the active release branches
+- Monitor CI signal, so a release can be cut at any time, and add CI signal for each new release branch
+- Maintain and improve user facing documentation about releases, release policy and calendar
+- Update the CAPI Netlify book certificates and DNS
+- Update the clusterctl Homebrew formula
+- Create and maintain the GitHub release milestone
+- Maintain and improve and release automation, tooling & related developer docs
+- Track tasks needed to add support for new Kubernetes versions in the next CAPI release
```

Comment:
Should we add something about managing CAPI version <=> Kubernetes version compatibility?

Not sure how to express it, but one example is: v1.25.0 was released last week, it's straightforward that v1.3 will support it, but what about v1.2.x. (consequences of the decision are that we have to cherry-pick certain PRs back into release-1.2 and ensure CI is configured accordingly)

Reply:
Let's defer that and all other k8s support tasks for now? as discussed in https://docs.google.com/document/d/1DQIsu2nLvkPD5x4c0GW7oYZjb1PUINv4mGPzB_PmPws/edit?disco=AAAAdQ5HoVE

Reply:
That's fine!